### DISCUSSION TITLE
Frage zu den Start / Stop Bedingungen des Dynamic Power Limiter

### DISCUSSION THREAD

**Discussion by 88markus88** [hoylabs/OpenDTU-OnBattery#562] [General]
Der Dynamic Power Limiter erlaubt es, die Start- und Stop Bedingungen für die Einspeisung aus der Batterie einzustellen. Wenn man einen Batteriemonitor / SmartShunt angeschlossen hat in % Füllstand mit
_Akku SoC - Start [%]_ und _Akkus SoC - Stop [%]_
Als Fallback für den Fall daß keine Daten vom SmartShunt durchkommen gibt es die Einstellungen:
_DC Spannung - Start [V]_ und _DC Spannung - Stop [V]_

Das funktioniert prima solange wie der SmartShunt tatsächlich die korrekten Ladezustandswerte in % überträgt. Dummerweise neigt ein Batteriemonitor dazu, langsam aber sicher größere Fehler aufzusummieren, wenn er nicht regelmäßig synchronisiert wird - normalerweise durch eine Vollladung. 
Es kann also passieren, daß der SmartShunt z.B. 90% Füllungsgrad anzeigt, während die Batterie nur noch 12.9[V] (Beispiel bezogen auf 4 Zellen LiFePO) hat, was real ca. 20% Füllungsgrad entspricht. 
Entlädt man jetzt die Batterie munter weiter, landet man beim Undervoltage Trigger des BMS, und sie wird abgeschaltet - während der SmartShunt immer noch behauptet daß der Füllungsgrad hoch ist (ca. 70% im Beispiel).

Jetzt meine Frage: Verhält sich onBattery tatsächlich so, oder gibt es sowas wie eine Sicherheitsabschaltung des Inverters wenn die Batteriespannung unter eine Grenze abfällt - auch wenn der SmartShunt noch genügend Ladezustand in% anzeigt?

**Comment by spcqike** [comment]
Ohne mir den Quellcode grade angeschaut zu haben, vermute ich, dass es sich wirklich so verhält. 

Wie schnell passiert der drift bei solchen großen Akkus denn? Beim Smartphone und co ist es ja bekannt. Da geht das Handy halt irgendwann mal bei 10-20% einfach aus und man lädt es einmalig auf 100% auf. 

Dem kannst du wahrscheinlich entgegenwirken, indem der Start threshold relativ hoch setzt und die „Leerung wenn voll“ nutzt.

**Reply by 88markus88** [reply]
Man weiß es halt nicht. Bei mir war die Diskrepanz nach einem Monat relativ hoch, weil meine ESPs aus dem Akku versorgt werden, und der Strom dafür unter der Detektionsschwelle meines BM liegt. 
Es wäre schon sinnvoll, wenn man die beiden Bedingungen "AND" verknüpfen könnte...

**Comment by schlimmchen** [comment]
Es ist definitiv so, wie ihr vermutet. Die Spannungsschwelle spielt nur dann eine Rolle, wenn der SoC nicht gemeldet wird.

Spannendes Thema, ich hab das nämlich auch erst lernen müssen: Auch mein JK BMS zeigte lange 40% SoC an, während der Akku langsam entleert wurde, bis die Spannung so weit abgesackt war, dass das BMS eingegriffen hat und der SoC dann plötzlich 0% war.

Allerdings bringt es meiner Einschätzung nach nichts, die untere Spannungsquelle noch zusätzlich zu berücksichtigen im DPL. Selbst wenn der DPL früher aufhört die Batterie zu entladen, so ist das Entladen durch den Laderegler in der Nacht, den Inverter (nur dessen Logik) in der Nacht, das BMS und ggf. dessen Display über längere Zeit das Problem. Und das hört nur auf, wenn man die Batterie abklemmt (was nach meiner Erfahrung aus diesem Winter meine zukünftige Strategie sein wird).

Oder übersehe ich etwas?

**Reply by 88markus88** [reply]
Ist schon ein Unterschied, ob der Inverter noch mit 200 Watt an der Batterie saugt, oder die ESPs/BM/BMS mit insgesamt 2 Watt (in meinem Fall): Mit laufendem Inverter landet man sehr leicht und schnell in der Notabschaltung durch das BMS, wenn onBattery keine verlässliche Info über den Ladezustand in % bekommt. Nur mit den "Kleinverbrauchern" werden bei mir ca. 1-2 Ah pro Tag verbraucht, die schaufelt das Solarpanel auch an einem trüben Tag noch rein. 

Ich sehe folgende generelle Lösungsmöglichkeiten:

1.  die vorgeschlagene "UND" Verknüpfung mit der Spannung. Das kann man dann so einstellen, daß die Stop-Spannung nur als Sicherheitsvorkehrung funktioniert.
2.  Ein separater Parameter "Safety-Voltage", bei dessen Unterschreiten der Inverter definitiv aus bleibt.
3. Ein (neuer) Mechanismus in der Software, der regelmäßig (z.B. einmal monatlich) den Akku über den Solarregler voll lädt, so daß der BM sich synchronisiert. 

(1) ist vermutlich am einfachsten machbar, daß das auch funktionieren muß wenn keim BM vorhanden ist, macht es etwas kompliziert. (2) ist besser, weil dann die Regelung nach Spannung unabhängig von der Abschaltung funktioniert. (3) wäre super, aber sehr aufwendig. Würde sich aber lohnen, denn auch im Sommer muß der BM regelmäßig synchronisiert werden, sonst wird der von ihm gelieferte Ladezustand in % mit der Zeit komplett falsch. Das ist prinzipbedingt, der BM zählt ja einfach nur die Coulomb rein/raus und bilanziert, das muß mit der Zeit ungenau werden.

**Reply by schlimmchen** [reply]
Aus etwas Entfernung und etwas abstrakt gesprochen scheinst du zu verlangen, dass OpenDTU-OnBattery schlauer ist als das BMS. Stimmst du mir zu?

In meinem Fall waren die Kleinverbraucher (BMS, BMS Display, Laderegler in der Nacht, Inverterlogik in der Nacht) insgesamt stärker als die Aufladung über die trüben und kalten Wochen Ende November bzw. Anfang Dezember.

1. Finde ich nachvollziehbar, aber wie gesagt: OpenDTU-OnBattery soll hier schlauer sein als das BMS. "Welchen SoC hat die Batterie?" "50%" "Aha, okay, aber du behauptest auch dass die Batterie nur 40V hat?! Ich glaub dir die 50% nicht. Die 40V glaube ich dir aber."
2. Die DPL Einstellungen sind schon sehr komplex, mehr Einstellungen helfen definitiv nicht.
3. Ich seh das ehrlich gesagt nicht als sinnvoll an. Vielleicht muss der Shunt im BMS oder SmartShunt kalibriert werden, aber eigentlich sollte da kein ernsthafter Drift entstehen?! Vielleicht bin ich da aber auch zu naiv und man muss ab und zu (wie oft denn?) einen vollen Ladezyklus erzwingen, selbst wenn man einen DoD von deutlich unter 100% anstrebt?!

**Reply by 88markus88** [reply]
@schlimmchen Mein Fokus ist  nicht das BMS der Batterie, sondern der Batteriemonitor, der ja parallel angeschlossen werden kann. Den braucht man vor allem wenn das BMS der Batterie nicht direkt ausgelesen werden kann, das ist bei mir tatsächlich der Fall (habe keine Pylontech).

Und ja, onBattery sollte da "schlauer" sein als (nur) den Ladezustand des Batteriemonitors auszulesen. Sonst landet man im Lauf der Zeit unweigerlich in einer Situation in der der BM aus der Synchronisation läuft, und in einer Notabschaltung des BMS. 
Eine Regelungseinrichtung wie on Battery muß so etwas vermeiden, nach meinen Maßstäben hat sie versagt wenn sie durch ein Sicherheitsfeature "gerettet" werden muß. Von daher: (1) oder (2) sollten schon umgesetzt werden.

Zu (3): Es ist leider so daß zählende Coulombmeter (= Shunt-basierter Batteriemonitor) **immer** aus der Synchronisation laufen. Das ist prinzipbedingt nicht zu vermeiden.  Je besser eingestellt desto langsamer, aber es passiert dennoch. Dehalb muß so ein BM regelmäßig synchronisiert werden, normalerweise durch eine Volladung die der BM erkennen kann (Endspannung, Schweifstom unterschritten). Entweder macht man das automatisch, das ist dann gut für die Nicht-Experten unter den Nutzern. Sonst eben manuell...

**Reply by spcqike** [reply]
> Dehalb muß so ein BM regelmäßig synchronisiert werden, normalerweise durch eine Volladung die der BM erkennen kann

das stimmt. Aber nur dann, wenn das BMS zuvor aufgrund von zu niedriger Spannung abgeschaltet hat. 

Der drift passiert dauerhaft und bei jeder Spannung. Wenn du von bspw 20% auf 100% auflädst, woher soll dass BMS wissen, dass die 20% nicht in Wirklichkeit nur 5% waren? Oder sogar 1%? Oder vielleicht sogar mehr?(letzteres wäre unwahrscheinlich, aber wenn kleine Entladungen wie 1-5W nicht erfasst werden, könnte kleine Ladeströme ebenfalls unerkannt bleiben) 

eine erneut korrekte Anzeige des SoC kann nur passieren, wenn sowohl ladestart als auch -ende definierte Punkte sind und die benötigte  Energie zum vollständigen aufladen gemessen werden konnte. 


wäre es nicht sinnvoll den lastausgang des Victron zu nutzen und den Punkt „battery life“ zu unterstützen? 


> 3.8.3. BatteryLife
Ist das Solarladegerät nicht in der Lage, die Batterie innerhalb eines Tages bis zu ihrer vollen Kapazität aufzuladen, wechselt der Status der Batterie ständig zwischen „partially charged“ (teilweise geladen) und „end of discharge“ (Ende der Entladung) hin und her. Dieser Betriebsmodus (kein regelmäßiges volles Aufladen) beschädigt eine Blei-Säure-Batterie binnen weniger Wochen oder Monaten.
Der BatteryLife Algorithmus überwacht den Ladezustand der Batterie und sofern erforderlich hebt er Tag für Tag den Schwellwert zum Abtrennen der Last an (d. h., die Last wird früher abgetrennt), bis die gewonnene Energie ausreicht, um die Batterie bis auf nahezu 100 % aufzuladen. Ab diesem Punkt wird der Schwellwert für das Abschalten der Last moduliert, so dass die Aufladung zu nahezu 100 % etwa einmal wöchentlich erreicht wird.

so dass die Batterie im Winter nicht zwischen nahe 0 und 10%, sondern eben zwischen 90-100% gehalten wird.
